sections/builder.js

```diff
@@ -1683,6 +1683,98 @@ export default [
       }
     ]
   },
+  {
+    type: "method",
+    method: "onConflict",
+    example: "insert(..).onConflict(column) / insert(..).onConflict([column1, column2, ...])",
+    description: "Utilized by PostgreSQL and Sqlite databases. When chained onto an insert query, it specifies the columns to be used in `ON CONFLICT` clause. A call to .onConflict should always be followed by either .ignore or .update (otherwise it does nothing). In MySQL this method is noop, and .ignore and .merge can be used without .onConflict.",
+    children: [
+      {
+        type: "text",
+        content: "Note: The column(s) specified by this method must have a UNIQUE index on them or the INSERT..ON CONFLICT query will fail to execute. When specifying multiple columns, they must have composite UNIQUE index."
+      },
+    ]
+  },
+  {
+    type: "method",
+    method: "ignore",
+    example: "insert(..).onConflict(..).ignore()",
+    description: "Implemented for the PostgreSQL, MySQL, and Sqlite databases. Modifies an insert query, and causes it to be silently dropped without an error if a conflict occurs. Uses INSERT IGNORE in MySQL, and adds an ON CONFLICT (columns) DO NOTHING clause to the insert statement in PostgreSQL and Sqlite.",
+    children: [
+      {
+        type: "runnable",
+        content: `
+          knex('tableName')
+            .insert({
+              email: "ignore@example.com",
+              name: "John Doe"
+            })
+            .onConflict('email')
+            .ignore()
+        `
+      }
+    ]
+  },
+  {
+    type: "method",
+    method: "merge",
+    example: "insert(..).onConflict(..).merge() / insert(..).onConflict(..).merge(updates)",
+    description: "Implemented for the PostgreSQL, MySQL, and Sqlite databases. Modifies an insert query, to turn it into an 'upsert' operation. Uses ON DUPLICATE KEY UPDATE in MySQL, and adds an ON CONFLICT (columns) DO UPDATE clause to the insert statement in PostgreSQL and Sqlite.",
```

Comment:
Even that the documentation says how the feature is implemented on each DB, this should also specify explicitly how to write crossdb compatible `.merge`. For many people from above it still will not be clear that mysql ignores the column list parameter and that actually to have cross db compatible merge, all the columns in the table must be listed.

Reply:
@nicoburns Could you please address this comment?

Reply:
@kibertoad Updated with a note on cross db compatibility.